src/java/org/apache/cassandra/db/guardrails/Values.java

```diff
@@ -0,0 +1,155 @@
+ */
+public class Values<T> extends Guardrail
+{
+    private final Function<ClientState, Config<T>> configProvider;
+    private final String what;
+
+    /**
+     * Creates a new values guardrail.
+     *
+     * @param configProvider a {@link ClientState}-based provider of {@link Config}s.
+     * @param what           The feature that is guarded by this guardrail (for reporting in error messages).
+     */
+    public Values(Function<ClientState, Config<T>> configProvider, String what)
+    {
+        this.configProvider = configProvider;
+        this.what = what;
+    }
+
+    /**
+     * Checks whether the provided value would trigger this guardrail.
+     *
+     * <p>This method is optional (does not have to be called) but can be used in the case some of the arguments
+     * to the actual guardrail method are expensive to build to save doing so in the common case of the guardrail not
+     * being triggered.
+     *
+     * @param value The value to test.
+     * @param state The client state, used to skip the check if the query is internal or is done by a superuser.
+     * @return {@code true} if {@code value} is not allowed by this guardrail,
+     * {@code false otherwise}.
+     */
+    public boolean triggersOn(T value, @Nullable ClientState state)
+    {
+        if (!enabled(state))
+            return false;
+
+        Config<T> config = configProvider.apply(state);
+        return config.getIgnored().contains(value) || config.getDisallowed().contains(value);
+    }
+
+    /**
+     * Triggers a warning if the provided value is disallowed by this guardrail and triggers an action to ignore it.
+     * If the value is disallowed it will abort the operation.
+     *
+     * @param value        The value to check.
+     * @param ignoreAction An action called on the subset of {@code values} that should be ignored. This action
+     *                     should do whatever is necessary to make sure the value is ignored.
+     * @param state        The client state, used to skip the check if the query is internal or is done by a superuser.
+     *                     A {@code null} value means that the check should be done regardless of the query.
+     */
+    public void guard(T value, Consumer<T> ignoreAction, @Nullable ClientState state)
+    {
+        if (!enabled(state))
+            return;
+
+        Config<T> config = configProvider.apply(state);
+
+        Set<T> ignored = config.getIgnored();
+        if (ignored.contains(value))
+        {
+            warn(format("Ignoring provided value %s as it is not supported for %s (ignored values are: %s)",
+                        value, what, ignored.toString()));
+            ignoreAction.accept(value);
+        }
+
+        Set<T> disallowed = config.getDisallowed();
+        if (disallowed.contains(value))
+            abort(format("Provided value %s is not allowed for %s (disallowed values are: %s)",
+                         value, what, disallowed.toString()));
+    }
```

Comment:
deadcode; can we remove?

Reply:
Removed. We'll probably add it back when we have a guardrail verifying a single value, like the one for allowed read/write consistency levels.